src/update_agent/actor.rs

```diff
@@ -143,11 +145,18 @@ impl UpdateAgent {
 
         let initialization = self.nop().map(|_r, actor, _ctx| {
             if actor.enabled {
-                log::info!("initialization complete, auto-updates logic enabled");
+                let status = "initialization complete, auto-updates logic enabled";
+                log::info!("{}", status);
+                update_unit_status(status);
+                notify_ready();
```

Comment:
Minor: seems like we could move those three lines to have the if-statement since they're in common with the else block? You'd have to declare `status` outside too.